pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/PreserveStackTraceRule.java

```diff
@@ -140,8 +141,12 @@ private boolean checkForTargetUsage(String target, Node baseNode) {
             List<ASTName> nameNodes = baseNode.findDescendantsOfType(ASTName.class);
             for (ASTName nameNode : nameNodes) {
                 if (target.equals(nameNode.getImage())) {
-                    match = true;
-                    break;
+                    boolean isPartOfStringConcatenation = 
+                        (nameNode.jjtGetParent().jjtGetParent().jjtGetParent() instanceof ASTAdditiveExpression);
```

Comment:
This might be a cornercase, but that code doesn't handle useless parentheses, eg: 
```java
throw new Exception("something bad:" + (e));
```

Reply:
Well pointed, updated